### PR TITLE
Update User Guide with Hyperlinks to Table of Contents; Update Income requirements for clairity

### LINKED ISSUE #140: User Guide Navigability
Add embedded links that allow the user to scroll to the top of the website, with just a click.

### LINKED ISSUE #153: Clients cannot be assigned a non-Integer value for income (e.g. 4.5)
image

### PR DESCRIPTION
**Pull request by colinhia**
Multiple edits to the User Guide, including:

Fixing of broken hyperlinks

Edits of income requirements for clarity

Closes #153. Closes #140.

### CODE REVIEW

**Review by FionaQY — APPROVED** [review]
LGTM

File: docs/UserGuide.md [hunk 1]
```diff
@@ -229,19 +235,19 @@ Arguments are the values that follow each flag in a command. **Arguments cannot
...
+| `e/`     | `<EMAIL>`             | The client's email address                                                                   | Valid email format (`username@domain.com`)                                                                                   | ❌                      |
+| `a/`     | `<ADDRESS>`           | The client's physical address                                                                | Any combination of letters, numbers, spaces, and symbols.                                                                    | ❌                      |
+| `j/`     | `<JOBNAME>`           | The client's job title or profession                                                         | Any combination of letters, numbers, spaces, and symbols.                                                                    | ❌                      |
+| `i/`     | `<INCOME>`            | The client's annual income                                                                   | Positive integer or zero <br/> • Must be numeric<br/> • Cannot include commas and decimal points<br/> • Cannot be fractional | ❌                      |
```
[the remark this answers is not in the document]
Comment (itsme-zeix) [reply docs/UserGuide.md:245]: I have no idea whether a layman will understand what's integer so I avoided using it, what do the others think?

File: docs/UserGuide.md [hunk 2]
```diff
@@ -63,6 +63,8 @@ AgentAssist is the **definitive desktop tool for credit card sales agents**. Mer
 
 Maximize your productivity, minimize your response time, and amplify your sales performance. With AgentAssist, you're not just keeping up with the competitive world of credit card sales — _you're setting the pace_.
 
+[↑ Return to Table of Contents](#table-of-contents)
```
Comment (itsme-zeix) [inline docs/UserGuide.md:66]: Thanks for the change!

**Review by itsme-zeix — APPROVED** [review]
LGTM! Great work~

**Comment by itsme-zeix** [comment]
This closes #140 as well!

File: docs/UserGuide.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (FionaQY) [reply docs/UserGuide.md:245]: If we want to cater to the layman, I vote positive number since this clarifies that commas and decimals are not allowed.

Comment (ZShunRen) [inline docs/UserGuide.md:245]:
```suggestion
| `i/`     | `<INCOME>`            | The client's annual income                                                                   | Positive number or zero <br/> • Must be numeric<br/> • Cannot include commas and decimal points<br/> • Cannot be fractional | ❌                      |
```

**Review by ZShunRen — APPROVED** [review]
LGTM

**Pull request merged**